### PR TITLE
Variables: Handle all sources with a new resolver

### PR DESCRIPTION
**Pull request by medikoo**
Closes: #8364

Accompanied by https://github.com/serverless/enterprise-plugin/pull/558 (ideally if it's merged right after this one, and released before we release new Framework version)

Configures all remaining variables sources in new variable resolver

### CODE REVIEW

**Review by pgrzesik — COMMENTED** [review]
Looks great overall :clap: I only have a few minor comments/questions, please let me know what do you think

File: docs/providers/aws/guide/plugins.md [hunk 1]
```diff
@@ -204,60 +204,62 @@ module.exports = Deploy;
...
-        serviceName: 'echo that isnt prepopulated',
-        isDisabledAtPrepopulation: true
-    };
-  }
-}
-```
 
-The above plugin will add support for variables like `${echo:foobar}` and resolve to the key. EG:
-`${echo:foobar}` will resolve to `'foobar'`.
+    this.configurationVariablesSources = {
```
[the remark this answers is not in the document]
Comment (medikoo) [reply docs/providers/aws/guide/plugins.md:218]: Yes, but that'll happen with a new major (no support is dropped with this PR)

it's dictated simply by fact that old variables resolver will be removed with next major, and same way old extensions will stop working.

However I don't think this feature is that used. It was created to aid dashboard plugin, and I'm not sure if any other plugins really benefited from that (and if they did migration will be really easy, you may check, how it looks in enterprise plugin PR)

[the remark this answers is not in the document]
Comment (pgrzesik) [reply docs/providers/aws/guide/plugins.md:218]: I just saw that in the related plugin PR - thanks for clarification - that's what I thought but wanted to make sure I'm not missing something here :+1:

File: lib/configuration/variables/sources/instance-dependent/get-cf.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        throw new ServerlessError(
+          'Missing address argument in variable "cf" source',
+          'MISSING_SLS_SOURCE_ADDRESS'
+        );
+      }
+      address = ensureString(address, {
+        Error: ServerlessError,
+        errorMessage: 'Non-string address argument in variable "cf" source: %v',
+      });
+      const separatorIndex = address.indexOf('.');
```
[the remark this answers is not in the document]
Comment (medikoo) [reply lib/configuration/variables/sources/instance-dependent/get-cf.js:21]: I thnk it's fine to assume then that `output.something` is target output name. Is such naming not supported by AWS?

File: docs/providers/aws/guide/plugins.md [hunk 3]
```diff
@@ -204,60 +204,62 @@ module.exports = Deploy;
...
+      foo: {
+        async resolve({ address, params, resolveConfigurationProperty, options  }) {
+          // `address` and `params` reflect values configured with a variable:
+          // ${foo(param1, param2):address}
+          // Note: they're passed if they're configured into variable
+
+          // `options` is CLI options
+          // `resolveConfigurationProperty` allows to access other confuguration properties,
```
[the remark this answers is not in the document]
Comment (medikoo) [reply docs/providers/aws/guide/plugins.md]: Good point, fixed

File: lib/configuration/variables/sources/instance-dependent/get-cf.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pgrzesik) [reply lib/configuration/variables/sources/instance-dependent/get-cf.js:21]: The output names have to be alphanumeric only and don't allow `.` in name, but I guess it's an edge case here

**Review by pgrzesik — APPROVED** [review]
Great job :raised_hands:

File: lib/configuration/variables/sources/instance-dependent/get-cf.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (medikoo) [reply lib/configuration/variables/sources/instance-dependent/get-cf.js:21]: Yes, I think we may leave this (at least at this point) to AWS validation. User should anyway be presented with meaningful message.

**Pull request merged**